### PR TITLE
Event table purger 

### PR DESCRIPTION
**Pull request by viktorshevchenko210**
**Describe the change**
Implemented Event table purger

**Describe your implementation or design**
Implemented EventsPurger in a similar way to WorkflowPurger.

**Tests**
Yes

**Breaking change**
No

**Additional context**
We had 16 million rows and it broke our PROD.

### CODE REVIEW

File: src/providers/WorkflowCore.Persistence.EntityFramework/Services/EventsPurger.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                db.Database.SetCommandTimeout(Options.DeleteCommandTimeoutSeconds);
+
+                #if NET6_0_OR_GREATER
+                    while(deleteEvents != 0)
+                    {
+                        deleteEvents = await db.Set<PersistedEvent>()
+                            .Where(x => x.EventTime < olderThanUtc &&
+                                        x.IsProcessed == true)
+                            .Take(Options.BatchSize)
+                            .ExecuteDeleteAsync(cancellationToken);
```
Comment (SergiiKram) [inline src/providers/WorkflowCore.Persistence.EntityFramework/Services/EventsPurger.cs:45]: Do we need `db.SaveChangesAsync` here?

File: test/WorkflowCore.IntegrationTests/Scenarios/WorkflowPurgeScenario.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            WaitForWorkflowToComplete(workflowId, TimeSpan.FromSeconds(30));
+
+            GetStatus(workflowId).Should().Be(WorkflowStatus.Complete);
+            UnhandledStepErrors.Count.Should().Be(0);
+            GetData(workflowId).StrValue1.Should().Be("Pass0");
+            GetData(workflowId).StrValue2.Should().Be("Pass1");
+
+            await WorkflowPurger.PurgeWorkflows(WorkflowStatus.Complete, DateTime.UtcNow);
+            await EventsPurger.PurgeEvents(DateTime.UtcNow);
+            await EventsPurger.PurgeEvents(DateTime.UtcNow);
```
Comment (SergiiKram) [inline test/WorkflowCore.IntegrationTests/Scenarios/WorkflowPurgeScenario.cs]: Why purge two times?

File: src/providers/WorkflowCore.Persistence.EntityFramework/Services/EventsPurger.cs (hunk 1 above)
Comment (viktorshevchenko210) [reply src/providers/WorkflowCore.Persistence.EntityFramework/Services/EventsPurger.cs:45]: No, we don't need 
https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-7.0/whatsnew#basic-executedelete-examples

File: test/WorkflowCore.IntegrationTests/Scenarios/WorkflowPurgeScenario.cs (hunk 2 above)
Comment (viktorshevchenko210) [reply test/WorkflowCore.IntegrationTests/Scenarios/WorkflowPurgeScenario.cs]: Changed

**Pull request merged**